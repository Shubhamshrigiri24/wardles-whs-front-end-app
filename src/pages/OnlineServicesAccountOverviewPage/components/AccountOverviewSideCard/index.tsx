import React, { FC, ReactNode, useState, useCallback } from "react";
import {
  Card,
  CardContent,
  makeStyles,
  Theme,
  useMediaQuery,
  List,
  ListItem,
  ListItemText,
  ListSubheader,
  Collapse,
} from "@material-ui/core";
import { Person, Phone, ExpandMore, LocationOn } from "@material-ui/icons";
import { useTheme } from "@material-ui/core/styles";
import clsx from "clsx";
import { TAccount } from "app/store/reducer/account/types";
import { wellColors, Typography } from "@welldigital/components";
import { formatDateFromIso } from "utils/formatters";

const useStyles = makeStyles((theme: Theme) => ({
  card: {
    borderRadius: 10,
    borderColor: wellColors.greyscale[500],
  },
  cardContentWrapper: {
    padding: "10px 16px !important",
    display: "flex",
    flexDirection: "column",
    height: "100%",
    position: "relative",
    [theme.breakpoints.up("md")]: {
      padding: "32px 24px !important",
    },
  },
  cardCollapseTrigger: {
    height: "24px",
    cursor: "pointer",
    display: "flex",
    alignItems: "center",
    justifyContent: "flex-end",
    position: "absolute",
    width: "100%",
    right: "0",
    padding: "0 16px",
    zIndex: 10,
  },
  expand: {
    transform: "rotate(0deg)",
    transition: theme.transitions.create("transform", {
      duration: theme.transitions.duration.shortest,
    }),
  },
  expandOpen: {
    transform: "rotate(180deg)",
  },
  cardList: {
    "&:not(:last-child)": {
      marginBottom: theme.spacing(2),
    },
  },
  cardListItem: {
    padding: "0 0 0 28px",
    margin: "0",
    "&:not(:last-child)": {
      marginBottom: theme.spacing(1),
    },
  },
  cardListSubHeader: {
    display: "flex",
    color: wellColors.elixir[900],
    marginBottom: "12px",
  },
  cardListSubHeaderText: {
    fontSize: "18px",
    fontWeight: 700,
    lineHeight: "150%",
  },
  cardListSubHeaderIcon: {
    marginRight: "4px",
  },
  cardListSubHeaderHasClick: {
    [theme.breakpoints.down("sm")]: {
      cursor: "pointer",
    },
  },
  cardListItemHeader: {
    color: wellColors.elixir[600],
  },
  cardListItemSubheader: {
    display: "block",
    fontWeight: 600,
    fontSize: "14px",
    color: wellColors.elixir[900],
    marginTop: theme.spacing(2),
    marginBottom: theme.spacing(1),
  },
  cardListItemText: {
    display: "block",
    fontWeight: 400,
    fontSize: "16px",
    color: wellColors.elixir[400],
    overflowWrap: "break-word",
  },
}));

export type AccountOverviewSideCardProps = {
  data: TAccount | null;
};

const AccountOverviewSideCard: FC<AccountOverviewSideCardProps> = ({
  data,
}) => {
  const classes = useStyles();
  const theme = useTheme();
  const isDesktop = useMediaQuery(theme.breakpoints.up("sm"), { noSsr: true });
  const [isOpen, setIsOpen] = useState<boolean>(false);

  const handleCollapse = useCallback(() => {
    setIsOpen(!isOpen);
  }, [isOpen]);
  return (
    <Card variant={"outlined"} classes={{ root: classes.card }}>
      <CardContent classes={{ root: classes.cardContentWrapper }}>
        {!data && (
          <Typography className={classes.cardListItemHeader} align={"center"}>
            No data
          </Typography>
        )}
        {data && (
          <>
            {!isDesktop && (
              <div
                className={classes.cardCollapseTrigger}
                onClick={handleCollapse}
              >
                <ExpandMore
                  className={clsx(classes.expand, {
                    [classes.expandOpen]: isOpen,
                  })}
                />
              </div>
            )}
            <Collapse
              in={isDesktop ? true : isOpen}
              timeout={"auto"}
              collapsedHeight={"24px"}
            >
              <List
                className={classes.cardList}
                component={"ul"}
                disablePadding
              >
                <ListSubheader
                  disableSticky
                  disableGutters
                  className={clsx(
                    classes.cardListSubHeader,
                    classes.cardListSubHeaderHasClick
                  )}
                >
                  <Person className={classes.cardListSubHeaderIcon} />
                  <Typography
                    className={classes.cardListSubHeaderText}
                    color={"inherit"}
                  >
                    Personal details
                  </Typography>
                </ListSubheader>
                <CardListItem
                  primary={`${data.firstName} ${data.lastName}`}
                  secondary={formatDateFromIso(data.dob, "dd/MM/yyyy")}
                />
              </List>
              <List
                className={classes.cardList}
                component={"ul"}
                disablePadding
              >
                <ListSubheader
                  disableSticky
                  disableGutters
                  className={classes.cardListSubHeader}
                >
                  <Phone className={classes.cardListSubHeaderIcon} />
                  <Typography
                    className={classes.cardListSubHeaderText}
                    color={"inherit"}
                  >
                    Contact details
                  </Typography>
                </ListSubheader>
                <CardListItem primary={"Email"} secondary={data.email} />
                <CardListItem primary={"Phone number"} secondary={data.phone} />
                <CardListItem
                  primary={"Registered address"}
                  secondary={`${data.addressline1} ${data.addressline2} ${data.postcode} ${data.city}`}
                />
              </List>
              <List
                className={classes.cardList}
                component={"ul"}
                disablePadding
              >
                <ListSubheader
                  disableSticky
                  disableGutters
                  className={classes.cardListSubHeader}
                >
                  <LocationOn className={classes.cardListSubHeaderIcon} />
                  <Typography
                    className={classes.cardListSubHeaderText}
                    color={"inherit"}
                  >
                    GP details
                  </Typography>
                </ListSubheader>
                <CardListItem primary={""} secondary={data.gpDetails} />
              </List>
            </Collapse>
          </>
        )}
      </CardContent>
    </Card>
  );
};

export type CardListItemProps = {
  icon?: ReactNode;
  primary?: string;
  secondaryHeader?: string;
  secondary?: string;
};

const CardListItem: FC<CardListItemProps> = ({
  primary,
  secondary,
  secondaryHeader,
}) => {
  const classes = useStyles();
  return (
    <ListItem
      className={classes.cardListItem}
      component={"li"}
      alignItems={"flex-start"}
      disableGutters
    >
      <ListItemText
        style={{ whiteSpace: "pre-line", margin: "0" }}
        primary={
          <Typography
            component={"span"}
            className={classes.cardListItemHeader}
            variant={"body2"}
          >
            {primary}
          </Typography>
        }
        secondary={
          <>
            {secondaryHeader && (
              <Typography
                component={"span"}
                className={classes.cardListItemSubheader}
              >
                {secondaryHeader}
              </Typography>
            )}
            <Typography component={"span"} className={classes.cardListItemText}>
              {secondary}
            </Typography>
          </>
        }
      />
    </ListItem>
  );
};

export default AccountOverviewSideCard;
