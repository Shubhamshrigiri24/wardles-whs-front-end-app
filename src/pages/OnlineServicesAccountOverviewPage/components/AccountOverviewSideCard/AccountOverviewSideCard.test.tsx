import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import AccountOverviewSideCard from "./index";

const data = {
  addressline1: "Nowhere street",
  addressline2: "",
  agreeMarketing: false,
  city: "Los Angeles",
  dob: "2000-04-02T00:00:00Z",
  email: "dummy@test.com",
  firstName: "Firstname",
  gender: "m",
  gpDetails: "Well GP Surgery",
  lastName: "Lastname",
  phone: "7777777",
  postcode: "M14LZ",
  updateTime: "0001-01-01T00:00:00Z",
  userId: "4352e499-8424-4a5e-8f0b-3a8d4f4a10b2",
};

describe("<AccountOverviewBanner />", () => {
  it("should match snapshot", () => {
    const { container } = render(<AccountOverviewSideCard data={data} />);
    expect(container).toMatchSnapshot();
  });
});
