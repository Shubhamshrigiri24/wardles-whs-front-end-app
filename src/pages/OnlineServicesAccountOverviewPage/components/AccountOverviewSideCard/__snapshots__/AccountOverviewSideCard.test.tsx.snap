// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccountOverviewBanner /> should match snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-card-1 MuiPaper-outlined MuiPaper-rounded"
  >
    <div
      class="MuiCardContent-root makeStyles-cardContentWrapper-2"
    >
      <div
        class="makeStyles-cardCollapseTrigger-3"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-expand-4"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
      </div>
      <div
        class="MuiCollapse-container"
        style="min-height: 24px;"
      >
        <div
          class="MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner"
          >
            <ul
              class="MuiList-root makeStyles-cardList-6"
            >
              <li
                class="MuiListSubheader-root makeStyles-cardListSubHeader-8 makeStyles-cardListSubHeaderHasClick-11"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root makeStyles-cardListSubHeaderIcon-10"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                  />
                </svg>
                <p
                  class="MuiTypography-root makeStyles-cardListSubHeaderText-9 MuiTypography-body1 MuiTypography-colorInherit"
                >
                  Personal details
                </p>
              </li>
              <li
                class="MuiListItem-root makeStyles-cardListItem-7 MuiListItem-alignItemsFlexStart"
              >
                <div
                  class="MuiListItemText-root MuiListItemText-multiline"
                  style="white-space: pre-line; margin: 0px;"
                >
                  <span
                    class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                  >
                    <span
                      class="MuiTypography-root makeStyles-cardListItemHeader-12 MuiTypography-body2"
                    >
                      Firstname Lastname
                    </span>
                  </span>
                  <p
                    class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                  >
                    <span
                      class="MuiTypography-root makeStyles-cardListItemText-14 MuiTypography-body1"
                    >
                      02/04/2000
                    </span>
                  </p>
                </div>
              </li>
            </ul>
            <ul
              class="MuiList-root makeStyles-cardList-6"
            >
              <li
                class="MuiListSubheader-root makeStyles-cardListSubHeader-8"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root makeStyles-cardListSubHeaderIcon-10"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6.62 10.79c1.44 2.83 3.76 5.14 6.59 6.59l2.2-2.2c.27-.27.67-.36 1.02-.24 1.12.37 2.33.57 3.57.57.55 0 1 .45 1 1V20c0 .55-.45 1-1 1-9.39 0-17-7.61-17-17 0-.55.45-1 1-1h3.5c.55 0 1 .45 1 1 0 1.25.2 2.45.57 3.57.11.35.03.74-.25 1.02l-2.2 2.2z"
                  />
                </svg>
                <p
                  class="MuiTypography-root makeStyles-cardListSubHeaderText-9 MuiTypography-body1 MuiTypography-colorInherit"
                >
                  Contact details
                </p>
              </li>
              <li
                class="MuiListItem-root makeStyles-cardListItem-7 MuiListItem-alignItemsFlexStart"
              >
                <div
                  class="MuiListItemText-root MuiListItemText-multiline"
                  style="white-space: pre-line; margin: 0px;"
                >
                  <span
                    class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                  >
                    <span
                      class="MuiTypography-root makeStyles-cardListItemHeader-12 MuiTypography-body2"
                    >
                      Email
                    </span>
                  </span>
                  <p
                    class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                  >
                    <span
                      class="MuiTypography-root makeStyles-cardListItemText-14 MuiTypography-body1"
                    >
                      dummy@test.com
                    </span>
                  </p>
                </div>
              </li>
              <li
                class="MuiListItem-root makeStyles-cardListItem-7 MuiListItem-alignItemsFlexStart"
              >
                <div
                  class="MuiListItemText-root MuiListItemText-multiline"
                  style="white-space: pre-line; margin: 0px;"
                >
                  <span
                    class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                  >
                    <span
                      class="MuiTypography-root makeStyles-cardListItemHeader-12 MuiTypography-body2"
                    >
                      Phone number
                    </span>
                  </span>
                  <p
                    class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                  >
                    <span
                      class="MuiTypography-root makeStyles-cardListItemText-14 MuiTypography-body1"
                    >
                      7777777
                    </span>
                  </p>
                </div>
              </li>
              <li
                class="MuiListItem-root makeStyles-cardListItem-7 MuiListItem-alignItemsFlexStart"
              >
                <div
                  class="MuiListItemText-root MuiListItemText-multiline"
                  style="white-space: pre-line; margin: 0px;"
                >
                  <span
                    class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                  >
                    <span
                      class="MuiTypography-root makeStyles-cardListItemHeader-12 MuiTypography-body2"
                    >
                      Registered address
                    </span>
                  </span>
                  <p
                    class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                  >
                    <span
                      class="MuiTypography-root makeStyles-cardListItemText-14 MuiTypography-body1"
                    >
                      Nowhere street  M14LZ Los Angeles
                    </span>
                  </p>
                </div>
              </li>
            </ul>
            <ul
              class="MuiList-root makeStyles-cardList-6"
            >
              <li
                class="MuiListSubheader-root makeStyles-cardListSubHeader-8"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root makeStyles-cardListSubHeaderIcon-10"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
                  />
                </svg>
                <p
                  class="MuiTypography-root makeStyles-cardListSubHeaderText-9 MuiTypography-body1 MuiTypography-colorInherit"
                >
                  GP details
                </p>
              </li>
              <li
                class="MuiListItem-root makeStyles-cardListItem-7 MuiListItem-alignItemsFlexStart"
              >
                <div
                  class="MuiListItemText-root MuiListItemText-multiline"
                  style="white-space: pre-line; margin: 0px;"
                >
                  <span
                    class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                  >
                    <span
                      class="MuiTypography-root makeStyles-cardListItemHeader-12 MuiTypography-body2"
                    />
                  </span>
                  <p
                    class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                  >
                    <span
                      class="MuiTypography-root makeStyles-cardListItemText-14 MuiTypography-body1"
                    >
                      Well GP Surgery
                    </span>
                  </p>
                </div>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
